### ISSUE TITLE
[Feature Request] Ability to create several tables in one version

### ISSUE THREAD

**Issue by SaifurRahmanMohsin** [rainlab/builder-plugin#63]
As the title suggests, it would be nice if it’s possible to have several tables created in one version change itself. Also, I find it odd that there are no provisions to apply foreign keys and also unique constraint on columns. Those would be nice features to have.

**Comment by vojtasvoboda** [comment]
+1

**Comment by alekseybobkov** [comment]
Builder has known limitations and they are described in the plugin's documentation. Index management is planned. Multiple PHP scripts per migration is unlikely to be implemented in the near future. It requires a complex UI solution which would move our focus away from our primary goal - the content management system development. Builder is a helper tool, which has it's own limitations, but still doesn't limit your abilities to develop plugins. Everything that can't be implemented with Builder can be done manually.

**Comment by tsioukas** [comment]
Maybe a feature that merges versions will help on that. Yes it's e a little problem because when you develop a plugin sometimes you need to add/remove/change table fields.

**Comment by SaifurRahmanMohsin** [comment]
Yes then it doesn’t help reduce plugin development time. So far, builder added more time for me to build plugins. I have switched back to console scaffolding because of these:
1. I tried making a Db. The table has a foreign key but builder doesn’t support it. So had to go back and fix it.
2. Since the Db is foreign key based it needs both tables to be pushed in same version. So had to write that again. Ultimately, I ended up removing the builder_table_create_some_table name and switched to the standard create_some_table.php format since I felt it’s too long.
3. I uploaded my plugin to octobercms.com. Got an error that I need to specify plugin name. I realize octoberCMS backend parses Plugin.php. I created a Plugin.php and move the Plugin.yaml data to this in appropriate sections. It all started to work now. I feel having two files would only create an overhead in the long run to have two different possible places for configuration.
4. Currently, Builder sets model timestamps to false when it generates the models. Several of the models I built required timestamp and there isn’t a provision within Builder to add timestamps. So, I had to manually enter timestamps, come back to backend, refresh and only then I was able to see created_on and updated_on appear in the autocomplete in Models section.

On a completely different note, I also had one technical issue with building a plugin that shared same navigation menu. I thought if I specify the same navigation name it may merge (it doesn’t work that way). In the end, I transferred even my second plugin’s Plugin.yaml into a Plugin.php and used [this](https://octobercms.com/forum/post/howto-add-side-menu-items-to-october-menus-or-plugins#post-9073) method to add the sub menu. I appreciate the amount of work that has been put into this and find some features really interesting but I haven’t personally felt a big difference in accelerating my plugin development. I think Builder is more oriented to non-developers who want to get into OctoberCMS development than for developers as they are pretty good doing things using console scaffolding commands.
